test/suites/ekuiper/streams_rules_test.go

```diff
@@ -58,10 +71,86 @@ func TestStreamsAndRules(t *testing.T) {
 	utils.TestDeviceVirtualReading(t)
 
 	t.Run("check rule_log", func(t *testing.T) {
-		utils.Exec(t, `edgex-ekuiper.kuiper-cli getstatus rule rule_log`)
+		var ruleStatus RuleStatus
+		var body string
+
+		// waiting for readings to come from edgex to ekuiper
+		for i := 1; ; i++ {
+			time.Sleep(1 * time.Second)
+			stdout, _, err := utils.Exec(t, "edgex-ekuiper.kuiper-cli getstatus rule rule_log")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			startIndex := strings.Index(stdout, "{")
+
+			if startIndex < 0 {
+				t.Fatal("Error getting status of rule rule_log in JSON format")
+			} else {
+				body = stdout[startIndex:]
+			}
+
+			err = json.Unmarshal([]byte(body), &ruleStatus)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			t.Logf("Waiting for readings to come from edgex to ekuiper, current retry count: %d/60", i)
+
+			if i <= 60 && ruleStatus.SouceCount > 0 {
+				t.Logf("Readings are coming to ekuiper now")
+				break
+			}
+
+			if i > 60 && ruleStatus.SouceCount <= 0 {
+				t.Logf("Waiting for readings to come from edgex to ekuiper, reached maximum retry count of 60")
+				break
+			}
+		}
+
+		stdout, _, err := utils.Exec(t, "edgex-ekuiper.kuiper-cli getstatus rule rule_log")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		startIndex := strings.Index(stdout, "{")
+
+		if startIndex < 0 {
+			t.Fatal("Error getting status of rule rule_log in JSON format")
+		} else {
+			body = stdout[startIndex:]
+		}
+
+		err = json.Unmarshal([]byte(body), &ruleStatus)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		require.Greaterf(t, ruleStatus.LogCount, 0, "No readings have been published to log by ekuiper")
```

Comment:
Doesn't a LogCount of greater than 0 imply the source count of greater than 0? I think checking LogCount in the above loop may be sufficient.